src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.connection.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String TITLE_COLUMN = "title";
+    private static final String PRICE_COLUMN = "price";
+    private static final String INSERT_BOOK_SQL = "INSERT INTO book (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID_SQL = "SELECT * FROM book WHERE id = ?";
+    private static final String FIND_ALL_SQL = "SELECT * FROM book";
+    private static final String UPDATE_BOOK_SQL =
+            "UPDATE book SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BOOK_SQL = "DELETE FROM book WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(INSERT_BOOK_SQL, Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, book.getTitle());
+            statement.setObject(2, book.getPrice());
+            if (statement.executeUpdate() < 1) {
+                throw new RuntimeException("Inserted book is empty");
+            }
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+            return book;
```

Comment:
Use `PreparedStatement` over `Statement` for the `INSERT` operation. Also, remove the `Statement.RETURN_GENERATED_KEYS` as it's only needed if you're going to use the generated keys which you are doing correctly. However, you should not throw a `RuntimeException` here. It's better to return an Optional.empty() or a custom exception that can be handled appropriately.